src/Nethermind/Nethermind.Experimental.Abi/AbiType.cs

```diff
@@ -0,0 +1,255 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Text;
+using Nethermind.Int256;
+
+namespace Nethermind.Experimental.Abi;
+
+public static partial class AbiType
+{
+    public static readonly IAbi<UInt256> UInt256 = new()
+    {
+        Name = "uint256",
+        Read = (ref BinarySpanReader r) =>
+        {
+            var bytes = r.ReadBytes(32);
+            return new UInt256(bytes, isBigEndian: true);
+        },
+        Write = (ref BinarySpanWriter w, UInt256 v) =>
+        {
+            Span<byte> bytes = stackalloc byte[32];
+            v.ToBigEndian(bytes);
+            w.Write(bytes);
```

Comment:
Provide a method renting buffer from the `BinarySpanWriter`, a similar to https://learn.microsoft.com/en-us/dotnet/api/system.buffers.ibufferwriter-1.getspan?view=net-9.0#system-buffers-ibufferwriter-1-getspan(system-int32) . I'd name it `.GetAndAdvance` and you'd use it here to `v.ToBigEndian(w..GetAndAvance(32))`. No double copy.